redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -183,7 +197,13 @@ internal open class UIViewLazyList(
       }
 
       override fun scrollViewDidEndScrollingAnimation(scrollView: UIScrollView) {
+        val wasDoingProgrammaticScroll = isDoingProgrammaticScroll
         isDoingProgrammaticScroll = false
+        
+        // Notify the scrollProcessor as needed
+        if (wasDoingProgrammaticScroll) {
+          scrollViewDidScroll(scrollView)
```

Comment:
I know that this method [should not be called for programmatic scrolls](https://github.com/cashapp/redwood/blob/7925bec1d6b135be5975a1d54ee36b6a3381a3cd/redwood-lazylayout-widget/src/commonMain/kotlin/app/cash/redwood/lazylayout/widget/LazyListScrollProcessor.kt#L56), but without this call, the placeholders at the top of the scrollView remain in place after the scroll animation finishes, without being swapped out for rendered widgets.

Is there a better hook to call here? Ideally the placeholders would swap out for rendered widgets _without animation_ after the scroll finishes.

Reply:
cc @swankjesse

Reply:
Marginal improvement: rather than keeping track of `wasDoingProgrammaticScroll`, it's better to just use the native [scrollViewDidScrollToTop](https://developer.apple.com/documentation/uikit/uiscrollviewdelegate/1619382-scrollviewdidscrolltotop) hook (which is actually fired earlier than this hook).

This feels better, but the above concern (needing to get rid of the second animation) remains to be addressed.